flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/TableEnvironment.scala

```diff
@@ -133,12 +134,24 @@ abstract class TableEnvironment(val config: TableConfig) {
         registerTableInternal(name, tableTable)
       case e: StreamTableEnvironment =>
         val sTableTable = new TransStreamTable(table.getRelNode, true)
-        tables.add(name, sTableTable)
+        schema.addTable(name, sTableTable)
     }
 
   }
 
   /**
+    * Unregisters a [[Table]] in the TableEnvironment's catalog.
+    * Unregistered tables cannot be referenced in SQL queries anymore.
+    *
+    * @param name The name under which the table is registered.
+    */
+  def unregisterTable(name: String): Unit = {
```

Comment:
Do we want to return a boolean success value or throw an exception if the table does not exist?